crowbar_framework/app/models/service_object.rb

```diff
@@ -1888,9 +1888,8 @@ def skip_unready_nodes(bc, inst, new_elements, old_elements)
       shared_elements.each do |n|
         pre_cached_nodes[n] ||= Node.find_by_name(n)
         node = pre_cached_nodes[n]
-        next if node.nil?
         # skip if nodes are on ready or crowbar_upgrade state, we dont need to do anything
-        next if ["ready", "crowbar_upgrade"].include?(node.state)
+        next if node && ["ready", "crowbar_upgrade"].include?(node.state)
```

Comment:
This doesn't really matter, to be honest. The rest of the code when applying would ignore the deleted node anyhow.

I would drop that part of the change, but I don't mind too much.